homeassistant/components/template/switch.py

```diff
@@ -141,9 +153,10 @@ def template_switch_state_listener(entity, old_state, new_state):
         @callback
         def template_switch_startup(event):
             """Update template on startup."""
-            async_track_state_change(
-                self.hass, self._entities, template_switch_state_listener
-            )
+            if self._template is not None:
+                async_track_state_change(
+                    self.hass, self._entities, template_switch_state_listener
+                )
 
             self.async_schedule_update_ha_state(True)
```

Comment:
This is weird, we shouldn't schedule an update. WE should remove this line and instead in the setup phase it should be `async_add_entiites(entities, True)`

Reply:
As this would change the original component code and as I don't understand exactly what you suggest here, I prefer not to change this. If you still want me to change something, please explain in more detail. Of course, you can also push a quick change to this branch yourself.

Reply:
Oh I actually see now that this is called in the state change handler, it's fine there.